### PR TITLE
Fix NFT Frames

### LINKED ISSUE #1460: Goerli | Radio streaming not working in NFT-Museum (85, -8)
### **Explorer Alpha build versions:**

- **v0.13.0**

### **Issue Description:**
The radio streaming feature is non-functional in the NFT-Museum scene located at coordinates (85, -8). 

### **STR:** 

1. Use the `/world goerli` command in the chat.
2. Navigate to coordinates (85, -8).
3. Whwn in the scene you should be able to listen to the radio streaming.

### **Expected behavior:**

https://github.com/user-attachments/assets/201256af-52dd-4ad7-81fa-61ca292ab017

### **Current behavior:**

https://github.com/user-attachments/assets/2bbfbe70-dab2-4aa7-a27d-5f27424e730b

#### **Additional Notes:**
Windows 11. 12th Gen Intel(R) Core(TM) i7-12650H 2.30 GHz. nvidia geforce RTX

### LINKED ISSUE #1462: Goerli | NFT paintings oversized in Goerli NFT-Museum (85, -8) and Simple-NFT (82,-1)
### **Explorer Alpha build versions:**

- **v0.13.0**

### **Issue Description:**
In the NFT-Museum scene, the NFT paintings are noticeably larger than their counterparts in the Prod environment. 

### **STR:** 

1. Use the `/world goerli` command in the chat.
2. Navigate to coordinates (85, -8) in the NFT-Museum scene.
3. Walk through the first and second floors of the scene.
4. Compare the sizes of the NFT paintings with those in the Prod.
5. Navigate to coordinates (82,-1) Simple-NFT scene.

### **Expected behavior:**

NFT-Paintings-PROD

https://github.com/user-attachments/assets/a68489f6-020c-4014-bf68-e254c2503cfe

image

### **Current behavior:**

NFT-Paintings-EA

https://github.com/user-attachments/assets/bf195381-ef8c-49fe-8d97-fc969398dbb6

image

#### **Additional Notes:**
Windows 11. 12th Gen Intel(R) Core(TM) i7-12650H 2.30 GHz. nvidia geforce RTX

### PR DESCRIPTION
**Pull request by mikhail-dcl**
Fixes #1462 
Fixes #1460

Fixed null ref exception that led to the scene suspension

P.S. There is a mismatch in materials of frames so the visuals of the frames are not the same. If we want to adjust them (I am not sure about the original decision) it's worth taking an extra step in a separate ticket

### CODE REVIEW

**Comment by nearnshaw** [comment]
:o how weird, that's not how I saw the NFTs yesterday on 0.13
I saw them the right way up, but twice as big as in prod

I also noticed that the elevator and other interactive things stopped working. My theory is that the scene crashes when we try to load the NFTs that are gifs. And maybe it crashes at different times in different machines, leaving things frozen on different intermediate loading states (?)

**Comment by nearnshaw** [comment]
If the frames have different materials, I suppose it's worth creating another separate ticket, yes. I'd consider that less of a priority to fix

**Comment by mikhail-dcl** [comment]
There is indeed an exception connected to videos, It's mentioned on another ticket

```
NullReferenceException: Object reference not set to an instance of an object
DCL.Optimization.Pools.GameObjectPool`1[T].HandleGet (T component) (at Assets/DCL/PerformanceAndDiagnostics/Optimization/Pools/GameObjectPool.cs:64)
UnityEngine.Pool.ObjectPool`1[T].Get () (at <c5ed782439084ef1bc2ad85eec89e9fe>:0)
DCL.Optimization.Pools.GameObjectPool`1[T].Get () (at Assets/DCL/PerformanceAndDiagnostics/Optimization/Pools/GameObjectPool.cs:41)
DCL.SDKComponents.MediaStream.CreateMediaPlayerSystem.CreateMediaPlayerComponent (System.String url, System.Boolean hasVolume, System.Single volume) (at Assets/DCL/SDKComponents/MediaStream/Systems/CreateMediaPlayerSystem.cs:82)
DCL.SDKComponents.MediaStream.CreateMediaPlayerSystem.CreateMediaPlayer (Arch.Core.Entity& entity, System.String url, System.Boolean hasVolume, System.Single volume) (at Assets/DCL/SDKComponents/MediaStream/Systems/CreateMediaPlayerSystem.cs:68)
DCL.SDKComponents.MediaStream.CreateMediaPlayerSystem.CreateVideoPlayer (Arch.Core.Entity& entity, DCL.ECSComponents.PBVideoPlayer sdkComponent) (at Assets/DCL/SDKComponents/MediaStream/Systems/CreateMediaPlayerSystem.cs:61)
DCL.SDKComponents.MediaStream.CreateMediaPlayerSystem.CreateVideoPlayerQuery (Arch.Core.World world) (at Arch.System.SourceGenerator/Arch.System.SourceGenerator.QueryGenerator/CreateMediaPlayerSystem.CreateVideoPlayer(in Entity, PBVideoPlayer).g.cs:35)
DCL.SDKComponents.MediaStream.CreateMediaPlayerSystem.Update (System.Single t) (at Assets/DCL/SDKComponents/MediaStream/Systems/CreateMediaPlayerSystem.cs:46)
ECS.Abstract.BaseUnityLoopSystem.Update (System.Single& t) (at Assets/Scripts/ECS/Abstract/BaseUnityLoopSystem.cs:29)
Rethrow as EcsSystemException: [MEDIA_STREAM] (85, -8): [CreateMediaPlayerSystem]
ECS.Abstract.BaseUnityLoopSystem.Update (System.Single& t) (at Assets/Scripts/ECS/Abstract/BaseUnityLoopSystem.cs:35)
Arch.SystemGroups.ExecutionNode`1[T].Update (T& t, System.Boolean throttle) (at ./Library/PackageCache/com.arch.systemgroups@bd58275fb1/ExecutionNode.cs:78)
Arch.SystemGroups.CustomGroupBase`1[T].UpdateInternal (T& t, System.Boolean throttle) (at ./Library/PackageCache/com.arch.systemgroups@bd58275fb1/Groups/CustomGroupBase.cs:119)
Arch.SystemGroups.DefaultGroup`1[T].Update (T& t, System.Boolean throttle) (at ./Library/PackageCache/com.arch.systemgroups@bd58275fb1/Groups/DefaultGroup.cs:46)
Arch.SystemGroups.ExecutionNode`1[T].Update (T& t, System.Boolean throttle) (at ./Library/PackageCache/com.arch.systemgroups@bd58275fb1/ExecutionNode.cs:74)
Arch.SystemGroups.CustomGroupBase`1[T].UpdateInternal (T& t, System.Boolean throttle) (at ./Library/PackageCache/com.arch.systemgroups@bd58275fb1/Groups/CustomGroupBase.cs:119)
ECS.Groups.SyncedGroup.Update (System.Single& t, System.Boolean throttle) (at Assets/Scripts/ECS/Groups/SyncedGroup.cs:73)
Arch.SystemGroups.ExecutionNode`1[T].Update (T& t, System.Boolean throttle) (at ./Library/PackageCache/com.arch.systemgroups@bd58275fb1/ExecutionNode.cs:74)
Arch.SystemGroups.SystemGroup.Update (Arch.SystemGroups.UnityBridge.TimeProvider+Info& timeInfo) (at ./Library/PackageCache/com.arch.systemgroups@bd58275fb1/SystemGroup.cs:107)
UnityEngine.DebugLogHandler:LogException(Exception, Object)
DCL.Diagnostics.DebugLogReportHandler:LogExceptionInternal(EcsSystemException) (at Assets/DCL/PerformanceAndDiagnostics/Diagnostics/ReportsHandling/Handlers/DebugLogReportHandler.cs:89)
DCL.Diagnostics.ReportHandlerBase:LogException(EcsSystemException) (at Assets/DCL/PerformanceAndDiagnostics/Diagnostics/ReportsHandling/Handlers/ReportHandlerBase.cs:42)
DCL.Diagnostics.ReportHubLogger:LogException(EcsSystemException, ReportHandler) (at Assets/DCL/PerformanceAndDiagnostics/Diagnostics/ReportsHandling/ReportHubLogger.cs:110)
DCL.Diagnostics.ReportHub:LogException(EcsSystemException, ReportHandler) (at Assets/DCL/PerformanceAndDiagnostics/Diagnostics/ReportsHandling/ReportHub.cs:106)
SceneRunner.Scene.ExceptionsHandling.SceneExceptionsHandler:Handle(Exception, ReportData) (at Assets/Scripts/SceneRunner/Scene/ExceptionsHandling/SceneExceptionsHandler.cs:111)
SceneRunner.Scene.ExceptionsHandling.SceneExceptionsHandler:Handle(Exception, Type) (at Assets/Scripts/SceneRunner/Scene/ExceptionsHandling/SceneExceptionsHandler.cs:36)
Arch.SystemGroups.SystemGroup:Update(Info&) (at ./Library/PackageCache/com.arch.systemgroups@bd58275fb1/SystemGroup.cs:120)
Arch.SystemGroups.DefaultSystemGroups.SimulationSystemGroup:Update() (at ./Library/PackageCache/com.arch.systemgroups@bd58275fb1/DefaultSystemGroups/SimulationSystemGroup.cs:22)
ECS.Prioritization.PartitionedWorldsAggregate:TriggerUpdate() (at Assets/Scripts/ECS/Prioritization/PartitionedWorldsAggregate.cs:42)
```

File: Explorer/Assets/DCL/SDKComponents/NFTShape/Frames/MeshFrame.cs [hunk 1]
```diff
@@ -29,6 +29,11 @@ public class MeshFrame : AbstractFrame
         public void Awake()
         {
             renderer.EnsureNotNull();
+
+            // Enforce renderer scale for backward compatibility (hardcoded in unity-renderer)
+            renderer.transform.localScale = new Vector3(0.5f, 0.5f, 1);
+            ;
```
Comment (m3taphysics) [inline Explorer/Assets/DCL/SDKComponents/NFTShape/Frames/MeshFrame.cs:35]: Minor typo?

**Review by m3taphysics — APPROVED** [review]

**Review by m3taphysics — APPROVED** [review]

**Review by anicalbano — APPROVED** [review]
🟢 QA Reviewed

image

image

**Pull request merged**